module/VuFind/src/VuFind/Auth/Manager.php

```diff
@@ -581,8 +582,20 @@ public function userHasLoggedOut()
      * Checks whether the user is logged in.
      *
      * @return UserRow|false Object if user is logged in, false otherwise.
+     *
+     * @deprecated Use getIdentity() or getUserObject() instead.
      */
     public function isLoggedIn()
+    {
+        return $this->getUserObject();
+    }
+
+    /**
+     * Checks whether the user is logged in.
+     *
+     * @return UserRow|false Object if user is logged in, false otherwise.
```

Comment:
This thought did occur to me, but I didn't do it because my plan is to replace getUserObject with getUserEntity in the Doctrine branch, and use stricter and more logical typing in getUserEntity. There's no reason not to do it now other than laziness, though -- if you think it's worth the effort, I'm happy to make your suggested change, and it will reduce the analysis needed when we get to the entity transition. What do you think?

Reply:
Would getUserObject map to getUserEntity for BC? If so, I'd do the change here.

Reply:
Actually, I just remembered the _other_ thing I was thinking about: creating a \VuFind\Auth\UserInterface. The primary difference between a Laminas object and a Doctrine entity is that the Laminas object relies on public properties while the Doctrine entity replies on getters. But there's no reason we couldn't define an interface for the user properties and implement that interface in \VuFind\Db\Row\User. Then, we could make getUserObject return the interface type, switch all the property references to getters, and have a totally seamless transition later. I think that's probably the best road forward (unless you disagree), but then the question is whether we merge this as incremental progress and finish in a separate PR, or if we undertake that whole big project here.

Reply:
That's a great idea, but I'd be fine doing that separately.

Reply:
Yeah, I think that would be clearer. If you don't mind approving this as-is, then, I'll begin work on that follow-up in the very near future.